lua/telescope/actions/set.lua

```diff
@@ -212,6 +212,43 @@ action_set.scroll_results = function(prompt_bufnr, direction)
   action_set.shift_selection(prompt_bufnr, math.floor(speed) * direction)
 end
 
+--- Select the current previewer and open it in a new scratch buffer.
+---
+---@param prompt_bufnr number: The prompt bufnr
+---@param type string: The type of selection to make
+--          Valid types include: "default", "horizontal", "vertical", "tabedit"
+action_set.select_preview = function(prompt_bufnr, type, opts)
+  opts = opts or {}
+  opts.ft = utils.get_default(opts.ft, nil)
+
+  local picker = action_state.get_current_picker(prompt_bufnr) -- picker state
+  if picker.previewer.state == nil then
+    utils.notify("action_set.select_preview", {
+      msg = "There is no previewer open",
+      level = "ERROR",
+    })
+    return
+  end
+
+  -- copy previewer content in new buffer
+  local preview_bufnr = picker.previewer.state.bufnr
+  local lines = vim.api.nvim_buf_get_lines(preview_bufnr, 0, -1, false)
+  local new_bufnr = vim.api.nvim_create_buf(false, true)
+  vim.api.nvim_buf_set_lines(new_bufnr, 0, -1, true, lines)
+
+  -- set context mark for jumplist
+  pcall(vim.api.nvim_set_current_win, picker.original_win_id)
+  vim.cmd "normal! m'"
+
+  -- open new buffer
+  edit_buffer(action_state.select_key_to_edit_key(type), new_bufnr)
+  if opts.ft == nil then
+    vim.cmd "filetype detect"
```

Comment:
I'm don't think this works as intended; for instance, try to select the preview of any lua file and there won't be any highlighting.

This could be fixed by something like this

```lua
local entry = action_state.get_select_entry()
-- type as per function signature needs some renaming ;)
if type(entry.value) == "string" then
  vim.api.nvim_buf_set_name(bufnr, entry.value)
end
```

which would work most often. For full coverage, refer to what `action_set.edit` is doing to extract file path.

While this appropriately then detects the filetype, it's a "clone" (rightfully so) of the actual file and, for instance, an error will be raised when the actual file would be opened (changes to the clone do not affect the file). Not sure how to best handle this, maybe prefix the filename with a `_` or something? Thoughts?

Reply:
The "filetype detect" only works when vim looks in the content indeed. E.g.: git commit messages.
I did not want to mess with filenames and buffer naming as this action would be interesting for cmd outputs more than files.
But I'll give it a shot and will let you know.

Reply:
File ft detection fixed in https://github.com/nvim-telescope/telescope.nvim/pull/1987/commits/03e6367aaeb3df5f14f3774f4e230cc776089afb

I did not name the buffer as it was creating a bunch of errors ("buffer with this name already exist") and that a scratch-buffer is preferable for this usecase IMO (temporary, not the actual file).
Instead I used the filename/path as done in `action_set.edit` and called plenary's ft detect().

This seem to work fine for both files and cmd outputs  on my end.
What do you think ?